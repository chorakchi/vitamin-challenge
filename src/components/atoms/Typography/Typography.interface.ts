import { HTMLAttributes } from "react";

export type variant =
  | "headline_xxl"
  | "headline_xl"
  | "headline_l"
  | "headline_m"
  | "headline_s"
  | "headline_xs"
  | "body"
  | "caption"
  | "inherit";
export type componentEnum =
  | "a"
  | "button"
  | "div"
  | "p"
  | "span"
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "h5"
  | "h6"
  | "li";
export type align = "inherit" | "left" | "center" | "right" | "justify";

export interface TypographyProps extends HTMLAttributes<HTMLElement> {
  /** @default false */
  bold?: boolean;
  /**
   * Applies the theme typography styles.
   */
  /** @default 'body' */
  variant?: variant;
  /**
   * If true, the text will have a bottom margin.
   */
  gutterBottom?: boolean;

  /**
   * The component used for the root node. Either a string to use a HTML element or a component.
   */
  component?: componentEnum;
  as?: any;
  href?: string;
  /**
   * it's for time using this component as label of input.
   */
  htmlFor?: string;
}
