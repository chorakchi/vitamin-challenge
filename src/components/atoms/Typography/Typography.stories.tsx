import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Typography } from "./Typography";

export default {
  title: "Atoms/Typography",
  component: Typography,
  parameters: {
    docs: {
      description: {
        component:
          `Use typography to present your design and content as clearly and efficiently as possible.<br/>
          The Typography component makes it easy to apply a default set of font weights and sizes in your application.
                <br/>` +
          `<br/> <br/> <h5> Changing the semantic element</h5>` +
          " The Typography component uses the variantMapping prop to associate a UI variant with a semantic element. It's important to realize that the style of a typography component is independent from the semantic underlying element. <br/> You can change the underlying element for a one-off situation with the ```component``` prop <br/> <br/>",
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof Typography>;

const Template: ComponentStory<typeof Typography> = (args) => (
  <Typography {...args} />
);

export const Typography_ = Template.bind({});
Typography_.args = {
  children: "Lorem Ipsum is simply dummy text of the printing",
  bold: true,
  variant: "body",
};
