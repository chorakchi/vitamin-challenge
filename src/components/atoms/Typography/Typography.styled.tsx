import styled, { css } from "styled-components";
import { TypographyProps } from "./Typography.interface";

export const TypographyStyled = styled.div<TypographyProps>`
  font-weight: ${({ bold }) => (bold ? 700 : 300)};

  color: #8a8b8c;
  font-family: "Circular TT", Arial;
  ${({ variant }) => {
    switch (variant) {
      case "headline_xxl":
        return css`
          /* h1 */
          line-height: 2.25rem;
          font-weight: 700;
          font-size: 1.875rem;
        `;
      case "headline_xl":
        /* h2 */
        return css`
          line-height: 2.25rem;
          font-weight: 700;
          font-size: 1.875rem;
        `;
      case "headline_l":
        /* h3 */
        return css`
          line-height: 1.75rem;
          font-weight: 700;
          font-size: 1.5rem;
        `;
      /* h4 */
      case "headline_m":
        return css`
          line-height: 1.5rem;
          font-weight: 700;
          font-size: 1.25rem;
        `;
      case "headline_s":
        /* h5 */
        return css`
          line-height: 1.25rem;
          font-weight: 700;
          font-size: 1.25rem;
        `;
      case "headline_xs":
        /* h6 */
        return css`
          font-size: 1rem;
          font-weight: 700;
          line-height: 1.25rem;
        `;
      case "body":
        return css`
          font-size: 1rem;
          line-height: 1.5rem;
        `;
      case "caption":
        return css`
          font-size: 0.875rem;
          line-height: 1.25rem;
        `;
      case "inherit":
        return css`
          font-size: "inherit";
        `;
      default:
        return css`
          font-size: 1rem;
          line-height: 1.5rem;
        `;
    }
  }}
`;
