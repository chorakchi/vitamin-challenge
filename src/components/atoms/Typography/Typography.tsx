import { TypographyStyled } from "./Typography.styled";
import { TypographyProps } from "./Typography.interface";
export const Typography: React.FC<TypographyProps> = ({
  variant = "body",
  ...props
}: TypographyProps) => {
  return <TypographyStyled {...props} variant={variant} />;
};
