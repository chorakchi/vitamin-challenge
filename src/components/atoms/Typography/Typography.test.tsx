import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Typography } from "./Typography";

describe("Typography", () => {
  it("should render the Typography", () => {
    const subject = shallow(<Typography>Typography</Typography>);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Typography>Typography</Typography>).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
