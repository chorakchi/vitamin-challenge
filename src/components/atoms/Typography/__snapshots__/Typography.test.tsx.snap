// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render correctly and match to the Snapshot 1`] = `
<div
  className="Typographystyled__TypographyStyled-sc-i22hyo-0 iKdygM"
>
  Typography
</div>
`;
